infra/main.bicep

```diff
@@ -265,12 +273,52 @@ module searchRoleBackend 'core/security/role.bicep' = {
   scope: searchServiceResourceGroup
   name: 'search-role-backend'
   params: {
-    principalId: backend.outputs.identityPrincipalId
+    principalId: backend.outputs.principalId
+    roleDefinitionId: '1407120a-92aa-4202-b7e9-c0e197c71c8f'
+    principalType: 'ServicePrincipal'
+  }
+}
+
+module searchRoleOpenAi 'core/security/role.bicep' = if (authType == 'rbac') {
```

Comment:
Any new roles you create will also need to be assigned to the user's principalId, see "user roles" above.